examples/sycl/sklearn_sycl.py

```diff
@@ -0,0 +1,136 @@
+    X = np.array([[1., 2.], [1., 4.], [1., 0.],
+                  [10., 2.], [10., 4.], [10., 0.]])
+    with sycl_context('gpu'):
+        kmeans = KMeans(n_clusters=2, random_state=0, init=X[:2]).fit(X)
+    print("kmeans.labels_")
+    print(kmeans.labels_)
+    print("kmeans.predict([[0, 0], [12, 3]])")
+    print(kmeans.predict([[0, 0], [12, 3]]))
+    print("kmeans.cluster_centers_")
+    print(kmeans.cluster_centers_)
+
+
+def k_means_random():
+    print("KMeans init='random'")
+    X = np.array([[1., 2.], [1., 4.], [1., 0.],
+                  [10., 2.], [10., 4.], [10., 0.]])
+    with sycl_context('gpu'):
+        kmeans = KMeans(n_clusters=2, random_state=0, init='random').fit(X)
+    print("kmeans.labels_")
+    print(kmeans.labels_)
+    print("kmeans.predict([[0, 0], [12, 3]])")
+    print(kmeans.predict([[0, 0], [12, 3]]))
+    print("kmeans.cluster_centers_")
+    print(kmeans.cluster_centers_)
+
+
+def linear_regression():
+    print("LinearRegression")
+    X = np.array([[1., 1.], [1., 2.], [2., 2.], [2., 3.]])
+    # y = 1 * x_0 + 2 * x_1 + 3
+    y = np.dot(X, np.array([1, 2])) + 3
+    with sycl_context('gpu'):
+        reg = LinearRegression().fit(X, y)
+    print("reg.score(X, y)")
+    print(reg.score(X, y))
+    print("reg.coef_")
+    print(reg.coef_)
+    print("reg.intercept_")
+    print(reg.intercept_)
+    print("reg.predict(np.array([[3, 5]]))")
+    print(reg.predict(np.array([[3, 5]])))
+
+
+def logistic_regression_lbfgs():
+    print("LogisticRegression solver='lbfgs'")
+    X, y = load_iris(return_X_y=True)
+    with sycl_context('gpu'):
+        clf = LogisticRegression(random_state=0, solver='lbfgs').fit(X, y)
+    print("clf.predict(X[:2, :])")
+    print(clf.predict(X[:2, :]))
+    print("clf.predict_proba(X[:2, :])")
+    print(clf.predict_proba(X[:2, :]))
+    print("clf.score(X, y)")
+    print(clf.score(X, y))
+
+
+def logistic_regression_newton():
+    print("LogisticRegression solver='newton-cg'")
+    X, y = load_iris(return_X_y=True)
+    with sycl_context('gpu'):
+        clf = LogisticRegression(random_state=0, solver='newton-cg').fit(X, y)
+    print("clf.predict(X[:2, :])")
+    print(clf.predict(X[:2, :]))
+    print("clf.predict_proba(X[:2, :])")
+    print(clf.predict_proba(X[:2, :]))
+    print("clf.score(X, y)")
+    print(clf.score(X, y))
+
+
+def dbscan():
+    print("DBSCAN")
+    X = np.array([[1., 2.], [2., 2.], [2., 3.],
+                  [8., 7.], [8., 8.], [25., 80.]])
+    with sycl_context('gpu'):
+        clustering = DBSCAN(eps=3, min_samples=2).fit(X)
+    print("clustering.labels_")
+    print(clustering.labels_)
+    print("clustering")
+    print(clustering)
+
+
+if __name__ == "__main__":
+    examples = [
+                k_means_init_x,
+                k_means_random,
+                linear_regression,
+                logistic_regression_lbfgs,
+                logistic_regression_newton,
+                dbscan,
+               ]
+    if gpu_available:
```

Comment:
It worth to add cpu context here so we will be validating that it's also possible to execute with CPU. and this will be validated in CI.

Reply:
done